lib/remap-lang/src/error.rs

```diff
@@ -105,3 +106,44 @@ impl fmt::Display for Rule {
         ]
     }
 }
+
+#[derive(Debug)]
+pub struct RemapError(pub(crate) Error);
+
+impl StdError for RemapError {
+    fn source(&self) -> Option<&(dyn StdError + 'static)> {
+        Some(&self.0)
+    }
+}
+
+impl fmt::Display for RemapError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
It seems like this could have some unfortunate effects if the source also does this recursive printing where you'd get the same error messages duplicated (consider if `source` is itself a `RemapError`).

I'm wondering if the `fmt::Display` implementation should be decoupled from the "error stack printing".

Not a big deal though, I don't think it is likely to cause issues. It just stood out to me.

Reply:
The source of `RemapError` is the concrete type `remap::Error`, so there should be no way for that to happen.

What could happen, is that the parent error that stores this error iterates over the sources, which indeed would produce duplicate error messages.

One way to prevent that is to "erase" the nested errors here, by returning `None` for `source()`, effectively making this the only error object to be returned by remap.

I'm not sure what the right approach is here, but I agree, it's unlikely to happen, and if it does, we can change it then.

I almost feel like our tracing macro's should do this by default, when we use `%error`.

Reply:
Gotcha, yeah, I see. This would only be an issue at a higher level. Agreed with punting for now.